docs/development/how_it_works.md

```diff
@@ -0,0 +1,132 @@
+# How it works
+
+## 1) The CLI
+
+Everything in `kops` is currently driven by a command line interface. We use [cobra](https://github.com/spf13/cobra) to define all of our command line UX.
+
+All of the CLI code for kops can be found in `/cmd/kops` [link](https://github.com/kubernetes/kops/tree/master/cmd/kops)
+
+For instance, if you are interested in finding the entry point to `kops create cluster` you would look in `/cmd/kops/create_cluster.go`. There you would find a function called `RunCreateCluster()`. That is the entry point of the command.
+
+## 2) Storage
+
+We have an abstracted way of interacting with our storage, called the `clientset`. This is an ambiguous way of interacting with remote storage for `kops`. This is commonly referred to as the `kops` **STATE STORE**.
```

Comment:
s/ambiguous/abstract

Maybe add a note that says that this could support e.g. server backends in future